rtmp-stream.go

```
@@ -18,32 +22,44 @@ const maxRtmpChannelNumber = 100
 var ErrorNoURLs = errors.New("No URLs available for streaming...")
 
 type RtmpStreamFactory struct {
-	URLs            []string
-	TimeoutDuration time.Duration
-	opened          int
+	hosts                []string
+	hostURLs             map[string][]*url.URL
+	TimeoutDuration      time.Duration
+	hostSelectionCounter int
+	opened               int
 }
 
-func (f *RtmpStreamFactory) nextURL() (string, error) {
-	urls := f.URLs
-	if len(urls) == 0 {
+func (f *RtmpStreamFactory) nextURL() (*url.URL, error) {
+	rand.Seed(time.Now().Unix())
+	for i := len(f.hosts); i >= 0; i-- {
+		if nextHost, er := f.nextHost(); er != nil {
```

error variables are called 'err' by convention.
Also, the error is discarded here - at least log it.